### DISCUSSION TITLE
Incomplete Advertising packet

### DISCUSSION THREAD

**Discussion by bbakerzzz** [micropython/micropython#9337] [Using MicroPython]
Hello all Thanks for all the help in advance.

I am kind of new to micropython 

My question/issue is when I use the LightBlue app on my phone I can see the following advertisement packet.
'''0201061AFF4C000215FDA50693A4E24FB1AFCFC6EB0764782500010002D80809616C6C77696E7311160318D0D5E036CE29000100020603E81B0000000000'''

When I use the following code snippet
'''def handle_scan(ev, data):
    addr_type, addr, adv_type, rssi, adv_data = data
    addr = hexlify(addr, ":").decode()
    adv_data = hexlify(adv_data, ":").decode()
    print(addr, adv_data)'''
I get
'''02:01:06:1a:ff:4c:00:02:15:fd:a5:06:93:a4:e2:4f:b1:af:cf:c6:eb:07:64:78:25:00:01:00:02:d8'''
Which match the begining of the packet.
'''0201061AFF4C000215FDA50693A4E24FB1AFCFC6EB0764782500010002D8'''

Which seems to be only part of the packet.  What do I need to do to change this behavior?

I am using the following setup
ESP32-wroom-32u
1.19.1 of micropython
The sending device is an Allwins Ibeacon

Again thanks for the help

**Comment by dpgeorge** [comment]
You may need to enable active scanning (See docs for `gap_scan`) and then the rest of the data might come in the active scan response.

**Reply by bbakerzzz** [reply]
I should have include how I start the scan.
'''
BLE().active(True)
BLE().irq(handle_scan)
BLE().gap_scan(60000, 55_000, 25_250) '''

Thanks but I did try that unless I have implemented the call incorrectly.  Any input would be greatly appreciated.

**Reply by jimmo** [reply]
See the `active` argument to gap_scan https://docs.micropython.org/en/latest/library/bluetooth.html#observer-role-scanner

Scanning is passive by default. If you enable active scanning, it will ask each device for it's "scan response" data, which enables an additional 31 bytes of data to be received.

**Reply by bbakerzzz** [reply]
Thanks for the information

When I use the following call.
'''BLE().gap_scan(20000, 55_000, 25_250, active='True')'''
I get the following error.
'''TypeError: function doesn't take keyword arguments'''

Do you know what I am doing wrong?

**Reply by bbakerzzz** [reply]
I have also tried without the quotes.

**Reply by bbakerzzz** [reply]
Never mind switched to and it is giving me more information.
'''BLE().gap_scan(20000, 55_000, 25_250, 1)'''

Again thanks for the information

**Comment by jimmo** [comment]
If you use aioble it will join these up for you.

**Reply by jimmo** [reply]
(The advertising payloads and the scan responses, that is)

**Reply by bbakerzzz** [reply]
Thanks for the answer

Not sure what you mean here can you point me to an example?
I will also research on my own.

**Reply by jimmo** [reply]
https://github.com/micropython/micropython-lib/tree/master/micropython/bluetooth/aioble

**Reply by puppet13th** [reply]
@bbakerzzz i recommend to use aioble module like @jimmo mentioned.

**Comment by straga** [comment]
How about BLE scan and WiFi sta - at the same time? For example scan BLE after sending to MQTT result.

**Reply by jimmo** [reply]
@straga Yes this should work.
